libraries/engage/filter/providers/FilterPageProvider.jsx

```diff
@@ -0,0 +1,320 @@
+import React, {
+  useMemo, useState, useEffect, useCallback,
+} from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import debounce from 'lodash/debounce';
+import isEqual from 'lodash/isEqual';
+import { router } from '@shopgate/engage/core';
+import {
+  getFiltersByHash,
+} from '@shopgate/engage/filter';
+// eslint-disable-next-line no-unused-vars, import/named
+import Context, { APIFilter, RouteFilters } from './FilterPageProvider.context';
+import { buildInitialFilters, buildUpdatedFilters } from '../helpers';
+
+/**
+ * @param {Object} state The application state.
+ * @param {Object} props The component props.
+ * @returns {Object}
+ */
+const mapStateToProps = (state, props) => ({
+  filters: getFiltersByHash(state, props),
+});
+
+/**
+ * @param {Object} next The next component props.
+ * @param {Object} prev The previous component props.
+ * @returns {boolean}
+ */
+const areStatePropsEqual = (next, prev) => {
+  if ((!prev.filters && next.filters) || (!isEqual(prev.filters, next.filters))) {
+    return false;
+  }
+
+  return true;
+};
+
+/**
+ * The FilterPageProvider component provides all relevant data and callbacks to represent and modify
+ * the current state of the "filter" page.
+ * @param {Object} props Provider props
+ * @param {APIFilter[]} props.filters Array of available filters
+ * @param {RouteFilters} props.activeFilters Object with the active filters for a filtered product
+ * list
+ * @param {string} props.parentRouteId Id of the route with the product list that's supposed to be
+ * filtered
+ * @param {Function} [props.onApply] Callback invoked when users pressed the apply button
+ * @param {string} [props.categoryId] A category to be used for filter selection from Redux
```

Comment:
categoryId and searchPhrase are not in the params

Reply:
even they are not used inside the component, they are part of the component props and accessible. they are needed by the connector to select the filters (as mentioned in the comment). without the documentation in the props there would be no type hints.